core/security_group.go

```diff
@@ -0,0 +1,172 @@
+package core
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/krystal/go-katapult"
+)
+
+type SecurityGroup struct {
+	ID               string   `json:"id,omitempty"`
+	Name             string   `json:"name,omitempty"`
+	AllowAllInbound  bool     `json:"allow_all_inbound,omitempty"`
+	AllowAllOutbound bool     `json:"allow_all_outbound,omitempty"`
+	Associations     []string `json:"associations,omitempty"`
+}
+
+func (sg *SecurityGroup) Ref() SecurityGroupRef {
+	return SecurityGroupRef{ID: sg.ID}
+}
+
+// allows a reference to a security group
+type SecurityGroupRef struct {
+	ID string `json:"id,omitempty"`
+}
+
+func (sg SecurityGroupRef) queryValues() *url.Values {
+	v := &url.Values{}
+	v.Set("security_group[id]", sg.ID)
+
+	return v
+}
+
+type SecurityGroupCreateArguments struct {
+	Name             string    `json:"name,omitempty"`
+	Associations     *[]string `json:"associations,omitempty"`
+	AllowAllInbound  *bool     `json:"allow_all_inbound,omitempty"`
+	AllowAllOutbound *bool     `json:"allow_all_outbound,omitempty"`
+}
+
+type SecurityGroupUpdateArguments struct {
+	Name             string    `json:"name,omitempty"`
+	Associations     *[]string `json:"associations,omitempty"`
+	AllowAllInbound  *bool     `json:"allow_all_inbound,omitempty"`
+	AllowAllOutbound *bool     `json:"allow_all_outbound,omitempty"`
+}
+
+type SecurityGroupCreateRequest struct {
+	Organization OrganizationRef               `json:"organization"`
+	Properties   *SecurityGroupCreateArguments `json:"properties,omitempty"`
+}
+
+type SecurityGroupUpdateRequest struct {
+	SecurityGroup SecurityGroupRef              `json:"security_group"`
+	Properties    *SecurityGroupUpdateArguments `json:"properties,omitempty"`
+}
+
+type SecurityGroupsResponseBody struct {
+	Pagination     *katapult.Pagination `json:"pagination,omitempty"`
+	SecurityGroup  *SecurityGroup       `json:"security_group,omitempty"`
+	SecurityGroups []*SecurityGroup     `json:"security_groups,omitempty"`
+}
```

Comment:
These three structs don't need to be exported, as they're purely used for internal over the wire data handling.